### ISSUE TITLE
sgp4init doesn't set 

### ISSUE THREAD

**Issue by chrisrosner** [brandon-rhodes/python-sgp4#68]
I was trying to use sgp4 to construct tles from orbital parameters using latest version (2.12)

I am getting invalid tles back, however. This seems to be because export_tle expects epochyr and epochdays to be set, but they don't seem to be set by sgp4init.

Maybe this is user error, but I couldn't quite figure it out.

Here's the test I constructed (using the sgp4init example from the documentation):

`from sgp4.api import WGS72, Satrec
from sgp4.exporter import export_tle

satrec = Satrec()

satrec.sgp4init(
    WGS72,           # gravity model
    'i',             # 'a' = old AFSPC mode, 'i' = improved mode
    5,               # satnum: Satellite number
    18441.785,       # epoch: days since 1949 December 31 00:00 UT
    2.8098e-05,      # bstar: drag coefficient (/earth radii)
    6.969196665e-13, # ndot: ballistic coefficient (revs/day)
    0.0,             # nddot: second derivative of mean motion (revs/day^3)
    0.1859667,       # ecco: eccentricity
    5.7904160274885, # argpo: argument of perigee (radians)
    0.5980929187319, # inclo: inclination (radians)
    0.3373093125574, # mo: mean anomaly (radians)
    0.0472294454407, # no_kozai: mean motion (radians/minute)
    6.0863854713832, # nodeo: right ascension of ascending node (radians)
)

tle = export_tle(satrec)

assert tle[0][18:32] != '00000.00000000'
`

**Comment by brandon-rhodes** [comment]
The underlying problem is that the C++ library `sgp4init()` that this library wraps does not set those fields when `sgp4init()` is called — strictly speaking, then, the underlying library does not support the output of a correct TLE for a satellite object produced from raw elements and a raw `epoch` value.

If the `epochyr` and `epochdays` attributes were writable, then we could expand the documentation and show users how to set them. But at the moment the C++ structure members are locked down as read-only so that the library can enforce a few well-known and trusted operations (like `sgp4init()`) as they way they get set.

Since they are not writable, I think it makes sense to (gulp!) incur a bit more expense in `sgp4init()` by adding some custom code to set `epochyr` and `epochdays`. Users that don't need those values will see `sgp4init()` run slightly more slowly, but in return they get valid values for all date fields without needing to consult further documentation or perform extra steps.

Accordingly, I have just committed 80a2471ef5681ecdfb128ac0c14f915503689836 which sets the two extra fields in both the pure-Python and also the C++ wrapper code.

Could you try installing from GitHub and trying again, to see if it indeed fixes your problem?
```
pip install https://github.com/brandon-rhodes/python-sgp4/archive/master.zip
```

**Comment by chrisrosner** [comment]
Amazing! I'll try it out today.

**Comment by chrisrosner** [comment]
I am now getting epochs in the tles.

I cannot believe how responsive you were on this, thank you!

**Comment by brandon-rhodes** [comment]
> I am now getting epochs in the tles.

Can you confirm that the epochs are correct, or just that they're there? :)

**Comment by chrisrosner** [comment]
I will attempt that next.

I know it doesn't make any sense, but for this application I don't need precision.

I'll comment today with a test for correctness :)

**Comment by chrisrosner** [comment]
This is the pytest I came up with. Looks good to me, but I will admit I get a little confused about the different epochs 1949 & Julian.

```
import datetime
from sgp4.api import WGS72, Satrec, jday
from sgp4.exporter import export_tle

def test_epoch_round_trip():

    """
    To compute the “epoch” value, simply take a normal Julian date and
    subtract ``2433281.5`` days.
    """
    d = datetime.datetime(2020, 10, 13, 16, 17, 18)
    jd, fr = jday(d.year, d.month, d.day, d.hour, d.minute, d.second)
    epoch = jd + fr - 2433281.5

    year = d.year % 100
    day_of_year = d.timetuple().tm_yday # 287
    day_fraction = day_of_year + (d.hour*60*60+d.minute*60+d.second)/(24*60*60)
    expected_epoch = '{0:02}{1:03.11}'.format(year, day_fraction)

    satrec = Satrec()

    satrec.sgp4init(
        WGS72,           # gravity model
        'i',             # 'a' = old AFSPC mode, 'i' = improved mode
        5,               # satnum: Satellite number
        epoch,           # epoch: days since 1949 December 31 00:00 UT
        2.8098e-05,      # bstar: drag coefficient (/earth radii)
        6.969196665e-13, # ndot: ballistic coefficient (revs/day)
        0.0,             # nddot: second derivative of mean motion (revs/day^3)
        0.1859667,       # ecco: eccentricity
        5.7904160274885, # argpo: argument of perigee (radians)
        0.5980929187319, # inclo: inclination (radians)
        0.3373093125574, # mo: mean anomaly (radians)
        0.0472294454407, # no_kozai: mean motion (radians/minute)
        6.0863854713832, # nodeo: right ascension of ascending node (radians)
    )

    tle = export_tle(satrec)

    assert expected_epoch == '20287.67868056'
    actual_epoch = tle[0][18:32]
    assert actual_epoch == expected_epoch


```

**Comment by brandon-rhodes** [comment]
Yes, it's rather idiosyncratic for TLEs to use their own epoch! Thanks very much for double-checking the value. I will try to get a new sgp4 version out this week!
